python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -83,3 +84,127 @@ def forward(self, g):
             )
 
         return degree_embedding
+
+
+class BiasedMultiheadAttention(nn.Module):
+    r"""Dense Multi-Head Attention Module with Graph Attention Bias.
+
+    Compute attention between nodes with attention bias obtained from graph
+    structures, as introduced in `Do Transformers Really Perform Bad for
+    Graph Representation? <https://arxiv.org/pdf/2106.05234>`__
+
+    Parameters
+    ----------
+    feat_size : int
+        Feature size.
+    num_heads : int
+        Number of attention heads, by which attr:`feat_size` is divisible.
+    bias : bool, optional
+        If True, it uses bias for linear projection. Default: True.
+    attn_bias_type : str, optional
+        The type of attention bias used for modifying attention. Selected from
+        'add' or 'mul'. Default: 'add'.
+
+        * 'add' is for additive attention bias.
+        * 'mul' is for multiplicative attention bias.
+

```

Comment:
Also, we may follow the convention to add a parameter as `attn_drop`. See the updated design doc.